token/program-2022/src/extension/confidential_transfer/processor.rs

```diff
@@ -121,13 +162,25 @@ fn process_configure_account(
         return Err(TokenError::MintMismatch.into());
     }
 
-    Processor::validate_owner(
-        program_id,
-        &token_account.base.owner,
-        authority_info,
-        authority_info_data_len,
-        account_info_iter.as_slice(),
-    )?;
+    match elgamal_pubkey_source {
+        ElGamalPubkeySource::ProofInstructionOffset(_) => {
+            Processor::validate_owner(
+                program_id,
+                &token_account.base.owner,
+                authority_info,
+                authority_info_data_len,
+                account_info_iter.as_slice(),
+            )?;
+        }
+        ElGamalPubkeySource::ElGamalRegistry(elgamal_registry_account) => {
+            // if ElGamal registry was provided, then just verify that the owners of the
+            // registry and token accounts match, then skip the signature
+            // verification check
+            if elgamal_registry_account.owner != *authority_info.key {
+                return Err(TokenError::OwnerMismatch.into());
+            }
```

Comment:
You'll also need to check that `authority_info.key == token_account.base.owner`, otherwise you can provide someone else's registry!

Reply:
Oh yeah I think this should actually be `elgamal_registry_account.owner != token_account.base.owner`. The authority pubkey does not necessarily need to be provided in the list of accounts since it is not providing any more info and is not a signer. I removed the authority pubkey from the instruction, so let me know what you think.

Reply:
Looks great!